src/mongo/delegates/Tables.class.php

```diff
@@ -337,16 +478,51 @@ public function generateTableRows($tableType,$resource=null,$context=null)
             return null;
         }
 
-        // ensure both the ID field and the impactIndex indexes are correctly set up
-        $collection->ensureIndex(array('_id.r'=>1, '_id.c'=>1,'_id.type'=>1),array('background'=>1));
-        $collection->ensureIndex(array('value.'._IMPACT_INDEX=>1),array('background'=>1));
+        // ensure that the ID field, view type, and the impactIndex indexes are correctly set up
+        $collection->ensureIndex(
+            array(
+                '_id.r'=>1,
+                '_id.c'=>1,
+                '_id.type'=>1
+            ),
+            array(
+                'background'=>1,
+                "socketTimeoutMS"=>\Tripod\Mongo\Config::getInstance()->getMongoCursorTimeout()
+            )
+        );
+
+        $collection->ensureIndex(
+            array(
+                '_id.type'=>1
```

Comment:
Is this going to impact on the space we have available if we're adding indexes to add table specs?